### PR TITLE
Python3 ready

### PR DESCRIPTION
**Pull request by kba**
These changes will make hocr-tools ready for Python 3.4, as outlined in https://github.com/tmbdev/hocr-tools/issues/37

Note that the code is **not** fully compatible with Python 3.4, since this would require correcting all the `print` calls and I wanted to keep the source as close to its current state as possible. The `2to3` tool bundled with Python can fix the source code automatically however and the converted source runs fine in 2.7 and 3.4 (c.f. [travis.yml](https://github.com/kba/hocr-tools/blob/python3-ready/.travis.yml#L12)).

Most important changes:
* Don't use lxml's `fromstring`/`tostring` but let lxml handle the distinction of filename and `stdin`/`stdout`
* Use `io.BytesIO` instead of `cStringIO` (which works in 2.7 because there is no distinction between strings and binary data, but is conceptually flawed)
* Don't use `string.join(sep, list)` which is equivalent to `sep.join(list)` and has been since at least 2.5
* Don't raise strings  but instances of real exceptions. It's pythonic and makes the code cleaner IMHO. Using exceptions for CLI argument errors seems overkill though, I replaced those with a helpful message and `sys.exit(1)`

The only hack-ish solution is a workaround for a bug in reportlab that will make `hocr-pdf` break in 3.x: The invisible font is loaded from a base64-encoded gzipped blob. reportlab does accept file-like objects for the filename argument to `TTFont` and loads the font properly. It will fail to decode it back to a `latin1` string when saving though. I've opened an [upstream PR for that](https://bitbucket.org/rptlab/reportlab/pull-requests/33/catch-pdfdata-being-byte/diff) but in the meantime, wrapping `sys.stdout` will work around this and not interfere with existing code.

### CODE REVIEW

File: hocr-pdf [hunk 1]
```diff
@@ -128,12 +128,13 @@ pL26nlWN2K+W1LhRjxlVGKmRTFYVo7CiJug09E+GJb+QocMCPMWBK1wvEOfRFF2U0klK8CppqqvG
...
-  ttf = cStringIO.StringIO(zlib.decompress(base64.decodestring(font)))
+""".encode('latin1')
+  uncompressed = bytearray(zlib.decompress(base64.decodestring(font)))
+  ttf = io.BytesIO(uncompressed)
   pdfmetrics.registerFont(TTFont('invisible', ttf))
 
 if __name__ == "__main__":
   if len(sys.argv) == 1:
-    print("Usage: %s <imgdir>\n" % os.path.basename(sys.argv[0]))
+    print(("Usage: %s <imgdir>\n" % os.path.basename(sys.argv[0])))
```
Comment (zuphilip) [inline hocr-pdf]: This change seems unrelated to the commit message (and it is a `print` change, which you want to correct at the moment).

File: hocr-check [hunk 2]
```diff
@@ -71,7 +71,10 @@ if len(sys.argv)<1:
 optlist,args = getopt.getopt(sys.argv[1:],"o")
 nooverlap = (assoc('-o',optlist)=='')
 if len(args)>0: stream = open(args[0])
-elif len(args)>1: raise "can only check one file at a time"
+elif len(args)>1:
+    print("Can only check one file at a time")
+    print_usage()
+    sys.exit(1)
```
Comment (zuphilip) [inline hocr-check]: Is the `elif` case reachable at all? 

`¬(len(args)>0) && (len(args)>1)` = ??

Maybe, we should switch the order of these two cases...

File: hocr-extract-images [hunk 3]
```diff
@@ -13,7 +13,7 @@ def assoc(key,list):
 
 def get_text(node):
     textnodes = node.xpath('.//text()')
-    s = string.join([text for text in textnodes])
+    s = ''.join([text for text in textnodes])
```
Comment (zuphilip) [inline hocr-extract-images]: This sould be in the previous commit.

File: hocr-pdf (hunk 1 above)
Comment (kba) [inline hocr-pdf]: An oversight.

**Comment by zuphilip** [comment]
A lot of lines deals with parsing the arguments, which I would like to do with the argparse module anyway, see #20 . Should we do that maybe before here?

**Comment by kba** [comment]
#20 absolutely should do it but not before this. I tried not to change the interfaces and using argparse module could cause that. I tried not to change the logic unless necessary (and was necessary for the lxml.fromstring changes).

File: hocr-extract-images (hunk 3 above)
Comment (kba) [inline hocr-extract-images]: True, but it's a trivial change, juggling the commits to get this patch into the commit before is not. Can do if you think it's important.

File: hocr-check (hunk 2 above)
Comment (kba) [inline hocr-check]: True, it's dead code at the moment. I would not fix it in this branch, tried not to change the logic (even if wrong).

File: hocr-extract-images (hunk 3 above)
Comment (zuphilip) [inline hocr-extract-images]: I guess we can have it also in this commit here. The only reason that I see is that clean commits for exactly one thing are best if at some point we may have to revert some of them. Let us be optimistic, we don't have to do that (and even if, then Travis will show us an error here).

**Comment by zuphilip** [comment]
This PR looks good for me now.

**Comment by zuphilip** [comment]
Thank you very much! This was merged with f49c0eb.
